### ISSUE TITLE
RoomListItem.roomInfo().displayName() often null

### ISSUE THREAD

**Issue by ganfra** [matrix-org/matrix-rust-sdk#4051]
When building the room summaries, we got a lot of `RoomListItem.roomInfo().displayName()` being `null`.
Then after some time, they get updated, but only in memory. 
Indeed, when launching the app in airplane mode, we got `null` values again for most of the rooms, even if the app had all the names before killing the app. 

I've checked the code, and the persistence to disk is only made during a next sync in the room only if there is new event in the room. So if you never get updates for a room, his display name will never be saved to disk.

It'd be great to have the `displayName` computed synchronously and not async, so we don't get null display name while we have avatar, and so it'd be saved to disk directly.

**Comment by bnjbvr** [comment]
Thanks for opening an issue, good catch. Computing the `displayName` synchronously would require a deep overhaul of the compute-room-name logic, which relies on events stored in the database at the moment; we should likely make it more generic and rely on database-or-sync, and save it immediately (even if the computing is async).

**Comment by Hywan** [comment]
I can replicate the bug. I believe it's the root cause for https://github.com/element-hq/element-x-ios/issues/3365 and https://github.com/element-hq/customer-success/issues/405. I'm working on it.

**Comment by Hywan** [comment]
The problem is the following:

https://github.com/matrix-org/matrix-rust-sdk/blob/cabb345a1c5b2b1ae4936eb2b3383af0c312d0f3/crates/matrix-sdk-base/src/client.rs#L1287-L1300

For some room, they receive a single update, then their `RoomInfo` are persisted, and finally their names are computed.
If another update is triggered, the cached name will be persisted, otherwise it won't.

That's why, in offline, some rooms have nothing, because the name wasn't persisted.

**Comment by bnjbvr** [comment]
Yes, the solution always has been to use little processors for the sync response. Quoting past self:

> we should likely make it more generic and rely on database-or-sync, and save it immediately (even if the computing is async).
